### DISCUSSION TITLE
Transforming the exchange correlation energy.

### DISCUSSION THREAD

**Discussion by simonganne01** [pyscf/pyscf#2632] [Q&A] [answered]
Hello everyone, 

I am trying to calculate $\left<\phi_i|e_{xc}|\phi_j\right>$ where $\phi_i$ and $\phi_j$ are wannier orbitals. So I need to do a linear transformation wrt to the atomic obritals (or molecular orbitals, but that is just another linear transformation). Currantly i do it in the fllowing manner: 
1. I calculate $e_{xc}$ w.r.t. the atomic orbitals using `exc =cgw._scf.get_veff()-cgw._scf.get_j()`, which gives me a matrix with dimesions (nkpts, nao, nao)
2. After that I transform this matrix to the wannier basis according to the following formula: $\Sigma_{n=1}^{nao}\Sigma_{m=1}^{nao}c_{ni}^{\dagger}\left<\phi_n|e_{xc}|\phi_m\right>c_{mj} = \left<\phi_i|e_{xc}|\phi_j\right>$   
with the following code: `exc_wan = np.einsum("kmn, kmi, knj -> ij", exc, mo_coeff_wan.conj(), mo_coeff_wan)` resulting in a matrix (number of wannier orbitals, numbr of wannier orbitals).
However, I get inconsistent results if i compare tzvp and dzvp basis sets. My question now is: Am I correct to assume that the functions `exc =cgw._scf.get_veff()-cgw._scf.get_j()` result in the exchange correlation energy in terms of the atomic orbitals? If so, Is my transformation correct or do I need to take the overlap still into acount?

**Comment by susilehtola** [comment, marked as answer]
Hello,

> Am I correct to assume that the functions `exc =cgw._scf.get_veff()-cgw._scf.get_j()` result in the exchange correlation energy in terms of the atomic orbitals

The density functional exchange-correlation energy is a non-linear function and thus does not agree with the above expression. For example the LDA exchange energy density is $\propto \rho^{4/3}$ while the potential in `veff` is $\propto \rho^{1/3}$.

It would help if you included some literature references on what you are trying to achieve.

**Comment by simonganne01** [comment]
I am trying to calculate the Vxc in terms of my wannier orbitals in order to implement the cGW algorithm proposed by the following paper: https://doi.org/10.1143/JPSJ.79.112001
the energy correction is given on page 13, equation 70. but I need to evaluate this operator between two wannier orbitals.

**Comment by simonganne01** [comment]
I am sorry, I noticed I've wrongly written $e_{xc}$ instead of $v_{xc}$.  so I think the mistake in with the $\rho$ dependence is solved because of the derivative, right?
